### ISSUE TITLE
[Youtube] Downloading of automated captions is not working (or at least very unpredictable)

### ISSUE THREAD

**Issue by th4t-gi** [ytdl-org/youtube-dl#27315]
## Checklist

- [x] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2020.12.05**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
$ youtube-dl -v --list-subs https://www.youtube.com/watch?v=48sCx-wBs34
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '--list-subs', 'https://www.youtube.com/watch?v=48sCx-wBs34']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.12.05
[debug] Python version 3.7.1 (CPython) - Darwin-20.1.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] 48sCx-wBs34: Downloading webpage
[youtube] 48sCx-wBs34: Looking for automatic captions
WARNING: Couldn't find automatic captions for 48sCx-wBs34
48sCx-wBs34 has no automatic captions
Available subtitles for 48sCx-wBs34:
Language formats
zh       vtt, ttml, srv3, srv2, srv1
en-US    vtt, ttml, srv3, srv2, srv1
$ youtube-dl -v --list-subs https://www.youtube.com/watch?v=48sCx-wBs34
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '--list-subs', 'https://www.youtube.com/watch?v=48sCx-wBs34']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.12.05
[debug] Python version 3.7.1 (CPython) - Darwin-20.1.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] 48sCx-wBs34: Downloading webpage
[youtube] 48sCx-wBs34: Looking for automatic captions
Available automatic captions for 48sCx-wBs34:
Language formats
af       vtt, ttml, srv3, srv2, srv1
sq       vtt, ttml, srv3, srv2, srv1
am       vtt, ttml, srv3, srv2, srv1
ar       vtt, ttml, srv3, srv2, srv1
hy       vtt, ttml, srv3, srv2, srv1
...
yo       vtt, ttml, srv3, srv2, srv1
zu       vtt, ttml, srv3, srv2, srv1
Available subtitles for 48sCx-wBs34:
Language formats
zh       vtt, ttml, srv3, srv2, srv1
en-US    vtt, ttml, srv3, srv2, srv1
```

## Description

Hi, I think there is a bug in the way that youtube-dl retrieves if a youtube video has auto captions because if you run `--list-subs` on any video with automatic captions, the output is inconsistent. This is also probably related to atleast #27306 and #27292.

**Comment by qazwsdc** [comment]
yes it seem to be a bug.
I also met this issue today.

**Comment by digitalsimulator** [comment]
I have been facing the same problem for the past few days. I сan't download automatically generated subtitles using --wirte-auto-sub option, although in fact they are present. (youtube-dl version 2020.12.07 and earlier). But it seems like the issue isn't reproducing regularly. Sometimes I can download subtitles, sometimes I can't. And in some cases, even if it is the same video.

**Comment by th4t-gi** [comment]
Yeah, it's definitely some type of bug with checking if there are automated subtitles but the author of #27344 mentioned that it might not be a bug with youtube-dl but with the youtube Data API.

**Comment by digitalsimulator** [comment]
I agree. The problem may be that YouTube has changed something. Today I tried to download subtitles of a certain video many times, but with no success. I used versions 2020.12.07 and 2020.11.19. However, after a while I suddenly succeeded.

**Comment by eseyman** [comment]
For the record, I ran "youtube-dl --list-subs https://www.youtube.com/watch?v=pwcZF12yx4M" 50 times in a loop.
youtube-dl reported 42 times that the video had no subs. The other 8 times, it reported that automatic captions were available.

**Comment by jvschiavo** [comment]
I also tried downloading automatic subtitles for a video and it didn't work, though it reported previously that there are automatic subtitles available for that video (and there is). :T

So... subtitle downloading isn't working right now? Maybe this should be the title of the issue (and maybe a pin issue)?

**Comment by zxl777** [comment]
```
youtube-dl --write-auto-sub --sub-lang en --skip-download "https://www.youtube.com/watch?v=GWGbOjlJDkU"
[youtube] GWGbOjlJDkU: Downloading webpage
[youtube] GWGbOjlJDkU: Looking for automatic captions
[info] Writing video subtitles to: Who are you-GWGbOjlJDkU.en.vtt

```

I expect to use this command to download automatically generated subtitle file, but it always download the manual subtitles.

**Comment by RallyQ** [comment]
已经三天了，ydl无法下载自动生成的字幕，7号之前下载字幕都正常。

但是火狐用的几个月前更新的YouTube字幕下载脚本，下载字幕正常

**Comment by JRRodriguez88** [comment]
I can also confirm this issue. It started three days ago, and appears to be present on both the current and previous version of youtube-dl. I suspect Youtube have changed something on their API.

youtube-dl --list-subs https://www.youtube.com/watch?v=pwcZF12yx4M
[youtube] pwcZF12yx4M: Downloading webpage
WARNING: video doesn't have subtitles
[youtube] pwcZF12yx4M: Looking for automatic captions
WARNING: Couldn't find automatic captions for pwcZF12yx4M
pwcZF12yx4M has no automatic captions
pwcZF12yx4M has no subtitles

In short, youtube-dl is not able to detect and download automatically generated captions!

**Comment by RogerRordo** [comment]
> 已经三天了，ydl无法下载自动生成的字幕，7号之前下载字幕都正常。
> 
> 但是火狐用的几个月前更新的YouTube字幕下载脚本，下载字幕正常

可以先用着[我这个](https://gist.github.com/RogerRordo/937d37a6bd19c20f1ce094cfe4fb48ea)替换顶着

**Comment by th4t-gi** [comment]
@RogerRordo I'm quite unfamiliar with the inner workings of youtube-dl, but if the gist that you shared can fix this bug, could you open a PR?

**Comment by vritme** [comment]
Auto subs are not downloading in my case also.
Just downloaded youtube-dl.exe several minutes ago, so this is last version. youtube-dl.exe --version
2020.12.09

youtube-dl.exe --skip-download --write-auto-sub https://www.youtube.com/watch?v=PqbB07n_uQ4
[youtube] PqbB07n_uQ4: Downloading webpage
[youtube] PqbB07n_uQ4: Looking for automatic captions
WARNING: Couldn't find automatic captions for PqbB07n_uQ4

![auto_subs_exist](https://user-images.githubusercontent.com/42464402/101964580-b0138500-3c33-11eb-9916-d9f99cadca33.jpg)

**Comment by RallyQ** [comment]
> 
> 
> > 已经三天了，ydl无法下载自动生成的字幕，7号之前下载字幕都正常。
> > 但是火狐用的几个月前更新的YouTube字幕下载脚本，下载字幕正常
> 
> 可以先用着[我这个](https://gist.github.com/RogerRordo/937d37a6bd19c20f1ce094cfe4fb48ea)替换顶着

我用的ydl是win版exe文件

**Comment by RogerRordo** [comment]
> @RogerRordo I'm quite unfamiliar with the inner workings of youtube-dl, but if the gist that you shared can fix this bug, could you open a PR?

The problem is that the structure of `ytplayer.config` has changed, and it can no longer be explicitly obtained from the webpage. So the function `_get_ytplayer_config` used for getting automatic captions is not available any more.

[My code](https://gist.github.com/RogerRordo/937d37a6bd19c20f1ce094cfe4fb48ea) fetches automatic captions from another object `ytInitialPlayerResponse`. However, I don't know how to fix the function `_get_ytplayer_config` which is still being called elsewhere. So I don't want to open a PR to submit this incomplete code.

**Comment by pipizhang** [comment]
> > 已经三天了，ydl无法下载自动生成的字幕，7号之前下载字幕都正常。
> > 但是火狐用的几个月前更新的YouTube字幕下载脚本，下载字幕正常
> 
> 可以先用着[我这个](https://gist.github.com/RogerRordo/937d37a6bd19c20f1ce094cfe4fb48ea)替换顶着

It works! Awesome

**Comment by denizveizov** [comment]
I also cannot get auto-generated subtitles, I updated to the latest version today -2020.12.12 but the issue persists.

**Comment by iNoSec2** [comment]
For me the issue happened after the beginning of december, or at least i saw there is a bug at the start of december. im the author of an other issue related to subtitle [(#27292)and](https://github.com/ytdl-org/youtube-dl/issues/27292) you can see the latest **version 2020.12.12** doesnt dl the sub when **version2020.12.02** does so it seems to be related to youtube-dl too or youtube API changed between again.

**Comment by RogerRordo** [comment]
> For me the issue happened after the beginning of december, or at least i saw there is a bug at the start of december. im the author of an other issue related to subtitle [(#27292)](https://github.com/ytdl-org/youtube-dl/issues/27292) and you can see the latest **version 2020.12.12** doesnt dl the sub when **version2020.12.02** does so it seems to be related to youtube-dl too or youtube API changed between again.

I only started using youtube_dl in the last few days, so I have not tested the previous version. But I compared the related code of the two versions, and found no significant difference. I believe that the random running result of the previous version is most likely due to the transition of the youtube API change. You can compare [my code](https://github.com/ytdl-org/youtube-dl/pull/27388/files) with the webpage of any youtube video to understand the cause of the bug.

**Comment by RallyQ** [comment]
> 
> 
> > 已经三天了，ydl无法下载自动生成的字幕，7号之前下载字幕都正常。
> > 但是火狐用的几个月前更新的YouTube字幕下载脚本，下载字幕正常
> 
> 可以先用着[我这个](https://gist.github.com/RogerRordo/937d37a6bd19c20f1ce094cfe4fb48ea)替换顶着

用pip装了ydl，用你代码替换后果然能下载字幕了，谢谢！

**Comment by freebrowser1** [comment]
Here as well. With 2020.12.12

youtube-dl --list-subs "https://www.youtube.com/watch?v=_3fusAQT0-U"
[youtube] _3fusAQT0-U: Downloading webpage
WARNING: video doesn't have subtitles
[youtube] _3fusAQT0-U: Looking for automatic captions
WARNING: Couldn't find automatic captions for _3fusAQT0-U
_3fusAQT0-U has no automatic captions
_3fusAQT0-U has no subtitles

When I watch it, it does have subtitles.
